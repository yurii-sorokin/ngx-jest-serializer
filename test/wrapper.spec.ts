// tslint:disable: max-classes-per-file no-input-rename
import { initHostFixture } from './framework/host'
import {
  Directive,
  Component,
  Input,
  TemplateRef,
  ViewContainerRef,
  ComponentFactoryResolver,
  ComponentRef,
  OnInit,
  NgModule
} from '@angular/core'
import { CommonModule } from '@angular/common'

describe(`Common`, () => {
  @Component({
    selector: '[tstWrapperContainer]',
    template: `<div class="wrapper">
    <ng-container *ngTemplateOutlet="template"></ng-container>
    </div>`
  })
  class WrapperContainerComponent {
    @Input() template
    @Input() text
  }

  @Directive({
    selector: '[tstWrapper]'
  })
  class WrapperDirective implements OnInit {
    @Input('tstWrapper') text

    private wrapperContainer: ComponentRef<WrapperContainerComponent>

    constructor(
      private templateRef: TemplateRef<any>,
      private viewContainerRef: ViewContainerRef,
      private componentFactoryResolver: ComponentFactoryResolver
    ) { }

    ngOnInit() {
      const containerFactory = this.componentFactoryResolver.resolveComponentFactory(WrapperContainerComponent)
      this.wrapperContainer = this.viewContainerRef.createComponent(containerFactory)
      this.wrapperContainer.instance.template = this.templateRef
    }
  }
  @NgModule({
    imports: [CommonModule],
    declarations: [WrapperDirective, WrapperContainerComponent],
    entryComponents: [WrapperContainerComponent],
    exports: [WrapperDirective, WrapperContainerComponent],
  })
  class WrapperModule { }

  it(`should print only wrapped content`, () => {
    const fixture = initHostFixture({
      imports: [WrapperModule],
      props: {
        wrapperText: 'wrapperText',
        text: 'content'
      },
      template:
        `<ng-container *tstWrapper="wrapperText"
        >
          <div>{{text}}</div>
        </ng-container>`
    })

    expect(fixture).toMatchSnapshot()
  })
})
