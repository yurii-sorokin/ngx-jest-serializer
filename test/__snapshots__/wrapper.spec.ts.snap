// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common should print only wrapped content 1`] = `
<tst-host>
  <div
    tstWrapperContainer=""
  >
    <div
      class="wrapper"
    >
      
    
      
          
      <div>
        content
      </div>
      
        
      
    
    </div>
  </div>
</tst-host>
`;
